### PR TITLE
var -> const

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Summary

I've left some pages untouched (like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Details_of_the_Object_Model) because those need a deep, deep rewrite. For some other cases (especially demonstrating legacy syntax) I've also kept the `var` to make the context intact.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/memory_management/index.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ Internet Explorer 6 and 7 are known to have reference-counting garbage collector
 
 ```js
 var div;
```
Comment (teoli2003) [inline files/en-us/web/javascript/memory_management/index.md:158]:
```suggestion
let div;
```

File: files/en-us/web/javascript/reference/deprecated_and_obsolete_features/the_legacy_iterator_protocol/index.md [hunk 2]
```diff
@@ -30,30 +30,29 @@ An object is an legacy iterator when it implements a `next()` method with the fo
...
+    next: function(){
+      if (nextIndex < array.length) {
+        return array[nextIndex++];
+      else
+        throw new StopIteration();
     }
+  }
 }
 
 var it = makeIterator(['yo', 'ya']);
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/deprecated_and_obsolete_features/the_legacy_iterator_protocol/index.md:46]:
```suggestion
let it = makeIterator(['yo', 'ya']);
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]

File: files/en-us/web/javascript/memory_management/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/memory_management/index.md:158]: This is describing behavior of IE6/7. I would either remove it entirely, or keep the `var`.

Comment (teoli2003) [reply files/en-us/web/javascript/memory_management/index.md:158]: IE6/7? Let's remove it then.

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**